src/header/utils.js

```diff
@@ -31,7 +31,9 @@ export const getContentMenuItems = ({ studioBaseUrl, courseId, intl }) => {
   return items;
 };
 
-export const getSettingMenuItems = ({ studioBaseUrl, courseId, intl }) => {
+export const getSettingMenuItems = ({
+  studioBaseUrl, courseId, intl, includeAdvancedSettings = true,
+}) => {
```

Comment:
Instead of passing this multiple levels down, I think you should convert this regular function into a react hook.  Here is what you can do:

- rename this function to `useSettingsMenuItems`
- remove all parameters other than courseId
- You can then use 
   ```javascript
   const studioBaseUrl = getConfig().STUDIO_BASE_URL;
   const intl = useIntl();
   const { canAccessAdvancedSettings } = useSelector(getStudioHomeData);
   ...
   ```

This will make the code much cleaner and avoid sending around unnecessary params.

Reply:
I converted all user functions to hooks as well, just for the sake of consistency. Hope that's fine. But if not, I can convert only this one. What do you think?

Reply:
Looks good!